docs: Fix references to recursive seal of Mocks

mariocj89 opened the pull request:
The documentation references the concept of "submocks" when just referring to child mocks, simplify the docstring and update the example as it assigns the mock without a name, which makes it a valid child mock and the seal recurses into it.

This was misinterpreted at the time of the initial PR as in my tests I've always named all the mocks 😞 .

Happy to open an issue if you think it is worth it.

mariocj89 committed:
docs: Remove "sumock" concept and fix seal documentation

As the sealing of the mock hierarchy is based on whether the mock has
the previous one as a parent or not, remove references to the undefined
"submock" concept and explain `seal` just in term of child mocks.

pablogsal commented on Lib/unittest/mock.py:
```diff
@@ -2422,15 +2422,12 @@ def __set__(self, obj, val):
-    parent. If a mock is assigned to an attribute of an existing mock,
-    it is not considered a submock.
-
+    The operation seals the mock passed in and all its children recursively.
```
Can you add here an explanation of what "child mocks" are? The documentation does not clarify exactly what a child mock is and this can be confusing.

pganssle commented on Lib/unittest/mock.py:
```diff
@@ -2422,15 +2422,14 @@ def __set__(self, obj, val):
-
+    The operation seals the mock passed in, all the mocks generated
+    by accessing an attribute of the mock argument and all assigned
+    mocks without a name or a `spec` recursively.
```
Is this supposed to be RST or md? If it's RST I think it needs to be

```rst
``spec``
```

instead of:

```markdown
`spec`
```

I also find the sentence confusing and don't really understand what it means.

pganssle replied:
I think you can re-word like this:

> The operation recursively seals the mock passed in, meaning that the mock itself, any mocks generated by accessing one of its attributes, and all assigned mocks without a name or `spec` will be sealed.

mariocj89 committed:
docs: Fix documentation about submocks for `unittest.mock.seal`

The documentation shows an example where using a new mock prevents
`seal` from applying recursively but for that to work the `Mock`
instance needs to have a name or a spec.

The docs in `library/unittest.mock` have been updated to remove
confusing terms about submock and be explicit about the behavior
expected.

pganssle commented on Lib/unittest/mock.py:
```diff
@@ -2422,15 +2422,14 @@ def __set__(self, obj, val):
-
+    The operation recursively seals the mock passed in, meaning that
+    the mock itself, any mocks generated by accessing one of its attributes,
+    and all assigned mocks without a name or spec.
```
The "that" makes this sentence not make sense. I think you need to add "will be sealed" at the end of this.
